pandas/core/indexes/base.py

```diff
@@ -5736,6 +5743,9 @@ def map(self, mapper, na_action=None):
             # empty
             attributes["dtype"] = self.dtype
 
+        if self._is_numeric_index and is_numeric_dtype(new_values.dtype):
```

Comment:
why is this here? should this not be handled in the Index constructor?

Reply:
The Index constructor cannot return a `NumericIndex` (until pandas 2.0), but should return a Int64Index etc to stay backward compatible.

OTOH, if we accept that operations like this return Int64Index etc, several tests will have to be changed and having this in an if statement is just easier IMO, because when we remove the `_is_numeric_index ` attribute this will fail hard, so we'll know then it should be removed.